cli/compose/loader/loader.go

```diff
@@ -380,9 +393,13 @@ func resolveVolumePaths(volumes []string, workingDir string) error {
 }
 
 // TODO: make this more robust
-func expandUser(path string) string {
+func expandUser(path string, lookupEnv template.Mapping) string {
 	if strings.HasPrefix(path, "~") {
-		return strings.Replace(path, "~", os.Getenv("HOME"), 1)
+		home, ok := lookupEnv("HOME")
+		if !ok {
+			panic(errors.New("cannot expand '~', because the environment lacks HOME"))
+		}
```

Comment:
@dnephin What should we do with warning?
Keep unparsed '~' as-is? Or Ignore the entry?